packages/create-turbo/__tests__/cli.test.ts

```diff
@@ -38,70 +40,113 @@ describe("create-turbo cli", () => {
 
   afterAll(() => {
     jest.setTimeout(DEFAULT_JEST_TIMEOUT);
+    execSync("corepack enable", { stdio: "ignore" });
+  });
+
+  afterEach(() => {
     if (fs.existsSync(testDir)) {
       fs.rmSync(testDir, { recursive: true });
     }
   });
 
-  it("guides the user through the process", (done) => {
-    let cli = spawn("node", [createTurbo, "--no-install"], {});
-    let promptCount = 0;
-    let previousPrompt: string;
-    const messages: string[] = [];
-    cli.stdout.on("data", async (data) => {
-      let prompt = cleanPrompt(data);
+  it("default: guides the user through the process", async () => {
+    const cli = spawn("node", [createTurbo, "--no-install"], {});
 
-      if (
-        !prompt ||
-        prompt.startsWith(">>> TURBOREPO") ||
-        isSamePrompt(prompt, previousPrompt)
-      ) {
-        return;
-      }
+    const messages = await runInteractiveCLI(cli);
 
-      promptCount++;
-
-      switch (promptCount) {
-        case 1:
-          expect(prompt).toEqual(
-            ">>> Welcome to Turborepo! Let's get you set up with a new codebase."
-          );
-          break;
-        case 2:
-          expect(prompt).toEqual(
-            `? Where would you like to create your turborepo? (./${DEFAULT_APP_NAME})`
-          );
-          cli.stdin.write(keys.enter);
-          break;
-
-        case 3:
-          // Which package manager do you want to use?
-          // easy to change deployment targets.
-          expect(getPromptChoices(prompt)).toEqual(["npm", "pnpm", "yarn"]);
-          cli.stdin.write(keys.enter);
-          break;
-        case 4:
-          // Bootstrap info
-          expect(
-            prompt.startsWith(
-              ">>> Bootstrapped a new turborepo with the following:"
-            )
-          ).toBe(true);
-
-          break;
-      }
+    expect(messages[0]).toEqual(
+      ">>> Welcome to Turborepo! Let's get you set up with a new codebase."
+    );
 
-      previousPrompt = prompt;
-    });
+    expect(messages[1]).toEqual(
+      `? Where would you like to create your turborepo? (./${DEFAULT_APP_NAME})`
+    );
 
-    cli.on("exit", () => {
-      try {
-        done();
-      } catch (error) {
-        done(error);
-      }
-      return;
-    });
+    expect(getPromptChoices(messages[2])).toEqual(["npm", "pnpm", "yarn"]);
+
+    expect(
+      messages[3].startsWith(
+        ">>> Bootstrapped a new turborepo with the following:"
+      )
+    ).toBe(true);
+
+    const newPackageJSON = JSON.parse(
+      fs.readFileSync(path.join(testDir, "package.json")).toString()
+    );
+    expect(newPackageJSON.packageManager).toMatch(/^npm/);
+  }, 10000);
+
+  it("--use-npm: guides the user through the process", async () => {
```

Comment:
Are these easily loopable?

Reply:
probably! i was trying not to abstract too much here while getting tests passing. Now that I'm adding `install` cases, I may rearrange some more